### PR TITLE
Put restriction on kubeStateMetricsCore rbac creation

### PR DESCRIPTION
**Pull request by tibatko509**
#### What this PR does / why we need it:
When kubeStateMetricsCore is enabled, we added the possibility to avoid the rbac creation automatically 
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [ X] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/templates/kube-state-metrics-core-rbac.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.datadog.kubeStateMetricsCore.enabled }}
+{{- if and (eq .Values.datadog.kubeStateMetricsCore.enabled true) (eq .Values.datadog.kubeStateMetricsCore.rbac.create true) }}
```
Comment (clamoriniere) [inline charts/datadog/templates/kube-state-metrics-core-rbac.yaml:1]: why not?
{{- if and .Values.datadog.kubeStateMetricsCore.enabled .Values.datadog.kubeStateMetricsCore.rbac.create }}

Comment (tibatko509) [reply charts/datadog/templates/kube-state-metrics-core-rbac.yaml:1]: yes you’re right. I am going to make the change

Comment (tibatko509) [reply charts/datadog/templates/kube-state-metrics-core-rbac.yaml]: done @clamoriniere

**Review by jtappa — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]

**Review by clamoriniere — COMMENTED** [review]
bash on the CI error, it seems that the scrip `helm-doc` has not been executed after the chart version bump.

**Comment by clamoriniere** [comment]
Hi @tibatko509 
I have integrated this PR into https://github.com/DataDog/helm-charts/pull/748 which is now merged

**Pull request closed without merging**